frontend/public/components/factory/table.tsx

```diff
@@ -52,152 +39,27 @@ import {
-    if (!filters || !filters.selected || !filters.selected.size) {
-      return true;
-    }
-    return filters.selected.has(resource.kind);
-  },
-
-  'build-status': (phases, build) => {
-    if (!phases || !phases.selected || !phases.selected.size) {
-      return true;
-    }
-
-    const phase = build.status.phase;
-    return phases.selected.has(phase) || !_.includes(phases.all, phase);
-  },
-
-  'build-strategy': (strategies, buildConfig) => {
-    if (!strategies || !strategies.selected || !strategies.selected.size) {
-      return true;
-    }
-
-    const strategy = buildConfig.spec.strategy.type;
-    return strategies.selected.has(strategy) || !_.includes(strategies.all, strategy);
-  },
-
-  'route-status': (statuses, route) => {
-    if (!statuses || !statuses.selected || !statuses.selected.size) {
-      return true;
-    }
-
-    const status = routeStatus(route);
-    return statuses.selected.has(status) || !_.includes(statuses.all, status);
-  },
-
-  'catalog-status': (statuses, catalog) => {
-    if (!statuses || !statuses.selected || !statuses.selected.size) {
-      return true;
-    }
-
-    const status = serviceCatalogStatus(catalog);
-    return statuses.selected.has(status) || !_.includes(statuses.all, status);
-  },
+import { tableFilters } from './table-filters';
 
-  'secret-type': (types, secret) => {
-    if (!types || !types.selected || !types.selected.size) {
-      return true;
-    }
-    const type = secretTypeFilterReducer(secret);
-    return types.selected.has(type) || !_.includes(types.all, type);
-  },
-
-  'pvc-status': (phases, pvc) => {
-    if (!phases || !phases.selected || !phases.selected.size) {
-      return true;
-    }
-
-    const phase = pvc.status.phase;
-    return phases.selected.has(phase) || !_.includes(phases.all, phase);
-  },
-
-  // Filter service classes by text match
-  'service-class': (str, serviceClass) => {
-    const displayName = serviceClassDisplayName(serviceClass);
-    return fuzzyCaseInsensitive(str, displayName);
-  },
-
-  'cluster-operator-status': (statuses, operator) => {
-    if (!statuses || !statuses.selected || !statuses.selected.size) {
-      return true;
-    }
-
-    const status = getClusterOperatorStatus(operator);
-    return statuses.selected.has(status) || !_.includes(statuses.all, status);
-  },
-
-  'template-instance-status': (statuses, instance) => {
-    if (!statuses || !statuses.selected || !statuses.selected.size) {
-      return true;
-    }
-
-    const status = getTemplateInstanceStatus(instance);
-    return statuses.selected.has(status) || !_.includes(statuses.all, status);
-  },
+const rowFiltersToFilterFuncs = (rowFilters) => {
+  return rowFilters
+    .filter(f => f.type && _.isFunction(f.filter))
+    .reduce((acc, f) => ({ ...acc, [f.type]: f.filter }), {});
 };
 
-const getFilteredRows = (_filters, objects) => {
+const getAllTableFilters = (rowFilters) => ({
+  ...rowFiltersToFilterFuncs(rowFilters),
+  ...tableFilters,
```

Comment:
Switch the two around so that rowFilters takes priority?

Reply:
Agreed, will do.